boskos/boskos_test.go

```diff
@@ -0,0 +1,660 @@
+		}
+
+		if rr.Code == http.StatusOK {
+			var data common.Resource
+			json.Unmarshal(rr.Body.Bytes(), &data)
+			if data.Name != "res" {
+				t.Errorf("%s - Got res %v, expect res", tc.name, data.Name)
+			}
+
+			if c.Resources[0].Owner != "o" {
+				t.Errorf("%s - Wrong owner. Got %v, expect o", tc.name, c.Resources[0].Owner)
+			}
+		}
+	}
+}
+
+func TestRelease(t *testing.T) {
+	var testcases = []struct {
+		name      string
+		resources []common.Resource
+		path      string
+		code      int
+		method    string
+	}{
+		{
+			name:      "get",
+			resources: []common.Resource{},
+			path:      "?name=res&dest=d&owner=foo",
+			code:      http.StatusMethodNotAllowed,
+			method:    http.MethodGet,
+		},
+		{
+			name:      "no arg",
+			resources: []common.Resource{},
+			path:      "",
+			code:      http.StatusBadRequest,
+			method:    http.MethodPost,
+		},
+		{
+			name:      "missing name",
+			resources: []common.Resource{},
+			path:      "?dest=d&owner=foo",
+			code:      http.StatusBadRequest,
+			method:    http.MethodPost,
+		},
+		{
+			name:      "missing dest",
+			resources: []common.Resource{},
+			path:      "?name=res&owner=foo",
+			code:      http.StatusBadRequest,
+			method:    http.MethodPost,
+		},
+		{
+			name:      "missing owner",
+			resources: []common.Resource{},
+			path:      "?name=res&dest=d",
+			code:      http.StatusBadRequest,
+			method:    http.MethodPost,
+		},
+		{
+			name:      "no resource",
+			resources: []common.Resource{},
+			path:      "?name=res&dest=d&owner=foo",
+			code:      http.StatusNotFound,
+			method:    http.MethodPost,
+		},
+		{
+			name: "wrong owner",
+			resources: []common.Resource{
+				{
+					Name:  "res",
+					Type:  "t",
+					State: "s",
+					Owner: "merlin",
+				},
+			},
+			path:   "?name=res&dest=d&owner=foo",
+			code:   http.StatusUnauthorized,
+			method: http.MethodPost,
+		},
+		{
+			name: "no match name",
+			resources: []common.Resource{
+				{
+					Name:  "foo",
+					Type:  "t",
+					State: "s",
+					Owner: "merlin",
+				},
+			},
+			path:   "?name=res&dest=d&owner=merlin",
+			code:   http.StatusNotFound,
+			method: http.MethodPost,
+		},
+		{
+			name: "ok",
+			resources: []common.Resource{
+				{
+					Name:  "res",
+					Type:  "t",
+					State: "s",
+					Owner: "merlin",
+				},
+			},
+			path:   "?name=res&dest=d&owner=merlin",
+			code:   http.StatusOK,
+			method: http.MethodPost,
+		},
+	}
+
+	for _, tc := range testcases {
+		c := MakeFakeClient(tc.resources)
+		handler := handleRelease(c)
+		req, err := http.NewRequest(tc.method, "", nil)
+		if err != nil {
+			t.Fatalf("Error making request: %v", err)
+		}
+		u, err := url.Parse(tc.path)
+		if err != nil {
+			t.Fatalf("Error parsing URL: %v", err)
+		}
+		req.URL = u
+		rr := httptest.NewRecorder()
+		handler.ServeHTTP(rr, req)
+		if rr.Code != tc.code {
+			t.Errorf("%s - Wrong error code. Got %v, expect %v", tc.name, rr.Code, tc.code)
+		}
+
+		if rr.Code == http.StatusOK {
```

Comment:
Check that LastUpdate() is greater than before? alternatively do this just in the ranch_test.go